Content.Client/UserInterface/Controls/RadialContainer.cs

```diff
@@ -67,37 +67,77 @@ public RadialContainer()
     {
 
     }
-	
-    protected override void Draw(DrawingHandleScreen handle)
+
+    /// <inheritdoc />
+    protected override Vector2 ArrangeOverride(Vector2 finalSize)
     {
-		
         const float baseRadius = 100f;
         const float radiusIncrement = 5f;
-		
-        var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
+
+        var children = ReserveSpaceForHiddenChildren
+            ? Children
+            : Children.Where(x => x.Visible);
+
         var childCount = children.Count();
-		
-		// Add padding from the center at higher child counts so they don't overlap.
-		Radius = baseRadius + (childCount * radiusIncrement);
+
+        // Add padding from the center at higher child counts so they don't overlap.
+        Radius = baseRadius + (childCount * radiusIncrement);
```

Comment:
You still have the issue with the sector radii being dependent on the number of children in the container

Also, this is going to override anyone setting the radius of the container in XML code

It should be 'AdjustedRadius = Radius + (childCount * radiusIncrement);' or similar

Reply:
@chromiumboy yes but that was in original control design - i have no idea why that was made. Are you saying that it should not be here now and this was a mistake? Or that for this type of buttons it should not be used?

Reply:
Radius being overwritten sounds like it was an oversight by the original coder. It needs to be corrected

Reply:
Oh it wasn't you, i'm sorry! Yea, i'll change it so it'll use provided from property radius as BaseRadius. Also i'll need to adjust radius that most controls were using - they used smaller numbers as base valus. Otherwise they will dramatically change in size - and we don't want such sudden change to stumble users...